packages/react-hooks/src/useBlockInterval.ts

```diff
@@ -30,7 +30,7 @@ function calcInterval (api: ApiPromise): BN {
         ? api.consts.timestamp.minimumPeriod.mul(BN_TWO)
         : api.query.parachainSystem
           // default guess for a parachain
-          ? DEFAULT_TIME.mul(BN_TWO)
+          ? api.consts.aura?.slotDuration ?? DEFAULT_TIME.mul(BN_TWO)
```

Comment:
There is a runtime api ;) `slot_duration` for Babe and Aura.

Reply:
I did try with `await api.call.auraApi.slotDuration()`, but this cannot run inside a React hook. Ideas to address this are more than welcome 😄

Reply:
> There is a runtime api ;) `slot_duration` for Babe and Aura.

Indeed better, but that still seems a bit consensus-dependent to me :)  I would even make it independent of that. Regardless of babe or aura or Pow or "no-consensus", one api for the runtime to say what it "thinks" the block time should be.

Reply:
> Indeed better, but that still seems a bit consensus-dependent to me :) I would even make it independent of that. Regardless of babe or aura or Pow or "no-consensus", one api for the runtime to say what it "thinks" the block time should be.

This does sound very ideal, but it would still be chain dependent if I understand correctly? For example, subspace uses `api.consts.subspace?.expectedBlockTime` which is integrated within `calcInterval` in the hook. So there is still some messy manual logic that will have to be applied.